src/main/java/net/dzikoysk/funnyguilds/data/configs/PluginConfiguration.java

```diff
@@ -1100,13 +1100,19 @@
     @CustomKey("check-for-restricted-guild-names")
     public boolean checkForRestrictedGuildNames = false;
 
+    @Comment("Jesli ustawione na false, nazwy i tagi z list 'blacklist-guild-names', 'blacklist-guild-tags' beda niedozwolone.")
+    @Comment("Jesli ustawione na true, listy beda traktowane jako whitelist, czyli beda dozwolone tylko nazwy z 'blacklist-guild-names' i 'blacklist-guild-tags'.")
+    @Comment("Przydatne kiedy chcesz ograniczyc tworzenie np. do 2 gildii \"RED\", \"BLUE\"")
+    @CustomKey("whitelist")
+    public boolean whitelist = false;
+
     @Comment("Niedozwolone nazwy przy zakladaniu gildii")
-    @CustomKey("restricted-guild-names")
-    public List<String> restrictedGuildNames = Collections.singletonList("Administracja");
+    @CustomKey("blacklist-guild-names")
+    public List<String> blacklistGuildNames = Collections.singletonList("Administracja");
 
     @Comment("Niedozwolone tagi przy zakladaniu gildii")
-    @CustomKey("restricted-guild-tags")
-    public List<String> restrictedGuildTags = Collections.singletonList("TEST");
+    @CustomKey("blacklist-guild-tags")
+    public List<String> blacklistGuildTags = Collections.singletonList("TEST");
```

Comment:
Please do not change config keys when there is no real need for it.

Reply:
When whitelist = true, name *blacklist* make less sens than restricted so why did you change that?